ambry-router/src/main/java/com.github.ambry.router/DefaultCryptoService.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.router;
+
+import com.github.ambry.config.CryptoServiceConfig;
+import java.nio.ByteBuffer;
+import java.security.SecureRandom;
+import java.security.Security;
+import javax.crypto.Cipher;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Default {@link CryptoService} which is capable of encrypting or decrypting bytes based on the given key.
+ * As of now supports GCM only for enryption and decryption
+ */
+class DefaultCryptoService implements CryptoService<SecretKeySpec> {
+
+  private static final String GCM_CRYPTO_INSTANCE = "AES/GCM/NoPadding";
+  private static final SecureRandom random = new SecureRandom();
+  private static final int ivValSize = 12;
+  private final Mode mode;
+  private final CryptoServiceConfig config;
+
+  private static final Logger logger = LoggerFactory.getLogger(DefaultCryptoService.class);
+
+  DefaultCryptoService(CryptoServiceConfig cryptoServiceConfig) {
+    this.config = cryptoServiceConfig;
+    Security.addProvider(new BouncyCastleProvider());
+    switch (config.cryptoServiceEncryptionDecryptionMode) {
+      case "GCM":
+        mode = Mode.GCM;
```

Comment:
I wanted to make it more generic to support more algos. Check out the ref here: https://github.com/bcgit/bc-java/blob/master/prov/src/test/java/org/bouncycastle/jce/provider/test/AESTest.java. But looks like some of them (atleast AES and CBC which I tried), have size difference between plain content size and decrypted content size. So, don't wanna spend too much time trying to make it work for everything. "GCM" is more secure than the rest. So, going with GCM for now and hence the supported modes are just "GCM"